share/ruby-build/3.1.0

```diff
@@ -1,2 +1,2 @@
-install_package "openssl-1.1.1n" "https://www.openssl.org/source/openssl-1.1.1n.tar.gz#40dceb51a4f6a5275bde0e6bf20ef4b91bfc32ed57c0552e2e8e15463372b17a" mac_openssl --if has_broken_mac_openssl
+install_package "openssl-1.1.1n" "https://www.openssl.org/source/openssl-1.1.1n.tar.gz#40dceb51a4f6a5275bde0e6bf20ef4b91bfc32ed57c0552e2e8e15463372b17a" openssl --if needs_openssl
```

Comment:
For CRuby 3.1+, OpenSSL 3 is supported, so there we should not install a custom openssl but use the system openssl.
So this could remain `--if has_broken_mac_openssl`.

I'm thinking we should have `needs_openssl3` (3.0 or older, so not 4+), `needs_openssl11` (1.1 or older, so not 3+), `needs_openssl10` (1.0 or older, so not 1.1+).
For CRuby 3.1+ we'd use `needs_openssl3`, for other CRuby versions which `install_package "openssl-1.1` it'd be `needs_openssl11` and for `install_package "openssl-1.0` it'd be `needs_openssl10`.
That could be done in this PR or later, but we should not install openssl for CRuby 3.1+ on non-macOS.